discord/client.py

```diff
@@ -117,6 +117,14 @@ class Client:
     r"""Represents a client connection that connects to Discord.
     This class is used to interact with the Discord WebSocket and API.
 
+    .. container:: operations
+
+        .. describe:: async with x
+
+            Asynchronously initialises the :class:`Client`.
```

Comment:
Probably something more like this
```suggestion
            Asynchronously initialises the client and automatically cleans up on scope exit.
```

Reply:
Should this be bot for Bot and AutoShardedBot?

Reply:
Yes. Also I hastily forgot that scope exit is a C++-ism so maybe just say "and automatically cleans up."